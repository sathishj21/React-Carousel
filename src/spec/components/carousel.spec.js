import React from 'react';
import { shallow } from 'enzyme';
import Carousel from '../../components/carousel';

describe('Carousel component', () => {
    const RendererComponent = (props) => {
        return shallow(<Carousel {...props} />);
    };

    const defaultProps = {};
    it('shallow renders Carousel component without crashing', () => {
        const CarouselComponent = RendererComponent({...defaultProps});
        expect(CarouselComponent).toMatchSnapshot();
    });

    it('test component scrollAction with target previous action instance', () => {
        const e = {
            target: {
                id: "prev"
            }
        };
        const component = RendererComponent({...defaultProps});
        const inst = component.instance();
        inst.scrollAction(e);
    });

    it('test component scrollAction with target previous action instance', () => {
        const e = {
            target: {
                parentNode: {
                    id: "next"
                }
            }
        };
        const component = RendererComponent({...defaultProps});
        const inst = component.instance();
        inst.scrollAction(e);
    });

    it('test component updateDimensions instance', () => {
        const component = RendererComponent({...defaultProps});
        const inst = component.instance();
        inst.refs = {carouselComponent: true};
        window.addEventListener = () => { };
        window.removeEventListener = () => { };
        inst.updateDimensions();
    });
});