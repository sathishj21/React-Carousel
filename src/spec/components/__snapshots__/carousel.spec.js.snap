// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component shallow renders Carousel component without crashing 1`] = `
<div
  className="main-container"
>
  <Loader />
</div>
`;
